server/session/session.go

```diff
@@ -79,6 +79,10 @@ type Session struct {
 	openChunkTransactions []map[uint64]struct{}
 	invOpened             bool
 
+	shouldSendSettingsResponse bool
```

Comment:
I would rename this and the field below it to something slightly better, maybe `hasServerSettings` and `serverSettings`

Reply:
These should also probably be atomic